### PR TITLE
Move msg []byte out of response

### PR DESCRIPTION
**Pull request by miekg**
Signed-off-by: Miek Gieben <miek@miek.nl>

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -542,15 +542,15 @@ func (srv *Server) serve(w *response, wg *sync.WaitGroup) {
...
 
-func (srv *Server) disposeBuffer(w *response) {
-	if w.udp != nil && cap(w.msg) == srv.UDPSize {
-		srv.udpPool.Put(w.msg[:srv.UDPSize])
+func (srv *Server) disposeBuffer(udp bool, m []byte) {
+	if udp && cap(m) == srv.UDPSize {
+		srv.udpPool.Put(m[:srv.UDPSize])
 	}
-	w.msg = nil
+	m = nil
```
Comment (tmthrgd) [inline server.go]: This isn’t needed anymore. It won’t block GC of the buffer as it’s only acessable during this function call.

**Comment by tmthrgd** [comment]
https://github.com/miekg/dns/pull/928#issuecomment-471159380
> #929 remove `msg` from response, this leads to a `nil` being passed to `serve()` which is somewhat ugly, but I can remember how that did look.

Originally it looked something like this, then I went and split up the code into two seperate `serveTCP` and `serveUDP` methods that were substantially cleaner. The worker code forced this all back through a single method. I’ll try and find my old PRs that cleaned this up.

File: server.go (hunk 1 above)
Comment (miekg) [reply server.go]: the `= nil` part you mean?

**Comment by miekg** [comment]
#930 because bases this of the *reverty* branch exceeded my git skills

**Pull request closed without merging**